### PR TITLE
Api 33481 poa v2 ind signatures

### PR DESCRIPTION
**Pull request by acovrig**
## Summary
- Require registrationNumber
- Validate regNum
- Add the signature to the generated PDF

## Related issue(s)
- [API-33481](https://jira.devops.va.gov/browse/API-33481)

## Testing done
- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## What areas of the site does it impact?
v2 individual POA

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/config/schemas/v2/2122a.json [hunk 1]
```diff
@@ -221,6 +221,7 @@
       "additionalProperties": false,
       "required": [
         "poaCode",
+        "registrationNumber",
```
Comment (jvcAdHoc) [inline modules/claims_api/config/schemas/v2/2122a.json:224]: we could probably remove first and last name from here. That would require adding them back into the form data like what you did for 2122.

File: modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122a/valid.json [hunk 2]
```diff
@@ -14,8 +14,7 @@
       },
       "representative": {
```
Comment (stiehlrod) [inline modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122a/valid.json:15]: @acovrig When I sent in the valid.json I get the following:
Screenshot 2024-03-28 at 1 39 10 PM

Comment (jvcAdHoc) [reply modules/claims_api/spec/fixtures/v2/veterans/power_of_attorney/2122a/valid.json:15]: Do you have a rep with that reigistration number and poa code in your local db?

File: modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb [hunk 3]
```diff
@@ -88,12 +88,21 @@ def organization_signatures(power_of_attorney, rep)
         }
       end
 
-      def individual_signatures(power_of_attorney)
-        first_name = power_of_attorney.form_data['representative']['firstName']
-        last_name = power_of_attorney.form_data['representative']['lastName']
+      def individual_signatures(power_of_attorney, rep)
         {
-          'page1' => individual_page1_signatures(power_of_attorney, first_name, last_name),
-          'page2' => individual_page2_signatures(power_of_attorney, first_name, last_name)
```
Comment (jvcAdHoc) [inline modules/claims_api/app/sidekiq/claims_api/v2/poa_form_builder_job.rb:96]: we can probably get rid of these page specific methods now

File: modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb [hunk 4]
```diff
@@ -70,7 +68,8 @@ def validate_registration_number!(form_number)
                                                          rn).order(created_at: :desc).first
           if rep.nil?
             raise ::Common::Exceptions::ResourceNotFound.new(
-              detail: "Could not retrieve Power of Attorney with registration number: #{rn} and poa code: #{poa_code}"
+              detail: "Could not find an Accredited Representative with registration number: #{rn} " \
```
Comment (jvcAdHoc) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb:71]: Nice, I like this language update

**Review by stiehlrod — APPROVED** [review]

**Pull request merged**